source/mainnet/tutorials/using-ID-in-dApps/wallet-connectors-tutorial.rst

```diff
@@ -0,0 +1,672 @@
+      console.error("Error:", error);
+    }
+  }, [provider]);
+
+This first ``useEffect`` hook:
+
+* Runs when the component mounts or when the provider changes
+* Returns a cleanup function that:
+
+  * Disconnects from the wallet
+  * Removes all event listeners to prevent memory leaks
+
+* The dependency array ``[provider]`` ensures this effect runs whenever the provider changes
+
+Next, let's look at the second ``useEffect`` that listens for account changes:
+
+.. code-block:: javascript
+
+  // Listen for account changes
+  useEffect(() => {
+    try {
+      const handleAccountChange = (newAccount: any) => {
+        setConnectedAccount(newAccount);
+      };
+
+      provider?.on("accountChanged", handleAccountChange);
+
+      return () => {
+        provider?.off("accountChanged", handleAccountChange);
+      };
+    } catch (error) {
+      console.error("Error:", error);
+    }
+  }, [provider]);
+
+This second ``useEffect`` hook:
+
+* Sets up an event listener for the provider's ``accountChanged`` event
+* When the account changes in the wallet, it updates our app's state
+* Also depends on the provider, so it runs whenever the provider changes
+
+Finally, let's render the wallet options:
+
+.. code-block:: jsx
+
+  return (
+    <Container className="connect-wallet-container text-center pt-2">
+      <h1 className="connect-wallet-title">Connect your wallet</h1>
+      {/* Browser Wallet Option */}
+      <Container
+        onClick={async (e) => {
+          connectProvider(await BrowserWalletProvider.getInstance());
+        }}
+        className="wallet-option p-4 cursor-pointer rounded-lg"
+      >
+        {/* UI elements for Browser Wallet */}
+      </Container>
+
+      {/* Mobile Wallet Option */}
+      <Container
+        onClick={async () => {
+          connectProvider(await WalletConnectProvider.getInstance());
+        }}
+        className="wallet-option p-4 rounded-lg cursor-pointer mt-2"
+      >
+        {/* UI elements for Mobile Wallet */}
+      </Container>
+    </Container>
+  );
+
+The render function creates a simple UI with:
+
+* A heading that prompts the user to connect a wallet
+* Two clickable containers representing the wallet options:
+
+  * Browser Wallet: Uses ``BrowserWalletProvider.getInstance()`` to get a provider instance
+  * Mobile Wallet: Uses ``WalletConnectProvider.getInstance()`` to get a provider instance
+
+* When clicked, each option calls the ``connectProvider`` function with the appropriate provider
+
+This component demonstrates a clean implementation pattern for wallet connections in React:
+
+1. It uses React Context from ``frontend/src/context/WalletContext.tsx`` to manage global wallet state
+2. It properly handles component lifecycle with ``useEffect``
+3. It sets up and cleans up event listeners to prevent memory leaks
+4. It provides a user-friendly interface for selecting a wallet type
+5. It encapsulates all the complexity of wallet connection behind a simple abstraction
```

Comment:
Please end full sentences with a `.` (occurs at several places in this tutorial).

```suggestion
1. It uses React Context from ``frontend/src/context/WalletContext.tsx`` to manage global wallet state.
2. It properly handles component lifecycle with ``useEffect``.
3. It sets up and cleans up event listeners to prevent memory leaks.
4. It provides a user-friendly interface for selecting a wallet type.
5. It encapsulates all the complexity of wallet connection behind a simple abstraction.
```